lib/bark.ex

```diff
@@ -2,56 +2,75 @@ defmodule Bark do
   require Logger
 
   # Logs a list of kv pairs
-  @spec warn(any(), Keyword.t()) :: any()
+  @spec warn(any(), Keyword.t() | binary()) :: any()
   def warn(env, opts), do: Logger.warn(parse_message(env, opts))
 
-  @spec info(any(), Keyword.t()) :: any()
+  @spec info(any(), Keyword.t() | binary()) :: any()
   def info(env, opts), do: Logger.info(parse_message(env, opts))
 
-  @spec error(any(), Keyword.t()) :: any()
+  @spec error(any(), Keyword.t() | binary()) :: any()
   def error(env, opts), do: Logger.error(parse_message(env, opts))
 
-  @spec debug(any(), Keyword.t()) :: any()
+  @spec debug(any(), Keyword.t() | binary()) :: any()
   def debug(env, opts), do: Logger.debug(parse_message(env, opts))
 
-  defp parse_message(env, opts) do
-    {function_name, arity} =
-      env
-      |> Map.get(:function)
-      |> case do
-        {function_name, arity} -> {function_name, arity}
-        _ -> {"None", 0}
-      end
-
-    module =
-      env
-      |> Map.get(:module)
-      |> case do
-        nil -> "None"
-        module -> module |> Atom.to_string() |> String.replace("Elixir.", "")
-      end
-
-    Enum.reduce(
-      opts,
-      "module=#{module} command=#{function_name}/#{arity}",
-      fn {k, v}, acc ->
-        "#{acc} #{Atom.to_string(k)}=#{log_value(v)}"
-      end
-    )
+  defp parse_message(env, opts) when is_list(opts) do
+    env
+    |> add_caller_context(opts)
+    |> to_log_string()
   end
 
+  defp parse_message(env, opts) when is_binary(opts) do
+    env
+    |> add_caller_context(message: opts)
+    |> to_log_string()
+  end
+
+  defp to_log_string(keywords) do
+    keywords
+    |> Enum.map(fn {key, value} ->
+      "#{Atom.to_string(key)}=#{log_value(value)}"
+    end)
+    |> Enum.join(" ")
+  end
+
+  defp add_caller_context(env, opts) when is_list(opts) do
+    opts
+    |> Keyword.put_new(:line, line_number(env))
+    |> Keyword.put_new(:command, function_name_arity(env))
+    |> Keyword.put_new(:module, module(env))
+  end
+
+  defp module(%{:module => module} = _env) when is_atom(module) do
+    module
+    |> Atom.to_string()
+    |> String.replace("Elixir.", "")
+  end
+
+  defp module(_), do: "None"
+
+  defp function_name_arity(%{:function => {function_name, arity}}),
+    do: "#{function_name}/#{arity}"
+
+  defp function_name_arity(_), do: "None"
+
+  defp line_number(%{line: line}), do: line
+  defp line_number(not_line), do: inspect(not_line)
+
   # Quote string if it contains a space
   defp quote_if_spaces(value) when is_binary(value) do
     if String.contains?(value, " ") do
-      "\"#{value}\""
+      value_with_escaped_quotes = String.replace(value, "\"", "\\\"")
```

Comment:
This is a new thing too, currently we're not escaping double quotes when they are being quoted.

Reply:
That's because it doesn't do anything in datadog, just makes it harder to read.

Reply:
Still seeing the same broken thing in the test for Grok Parser:
Normal no internal quotes
```
19:57:12.290 request_id=FfDoburTQOzShUwAAUQB testing="apples asdf"
{
  "timestamp": 1599249432290,
  "request_id": "FfDoburTQOzShUwAAUQB",
  "testing": "apples asdf"
}
```
With escaped quotes
```
19:57:12.290 request_id=FfDoburTQOzShUwAAUQB testing="apples \" asdf"
{
  "timestamp": 1599249432290,
  "request_id": "FfDoburTQOzShUwAAUQB",
  "testing": "apples \\"
}
```

Reply:
Maybe we should strip the quotes?  🤷

Reply:
Strip or what about replace with `_`  or ✊🏻 or 🩹?